File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.py
class Solution:
    def totalMoney(self, n: int) -> int:
        sum = 0
        no = 0
        mul = 1
        for num in range(1,n+1):
            if num <=7:
                sum +=num
            else:
                no = num % 7
                if no == 0:
                    sum += 7 + mul
                    mul +=1
                else:
                    sum += (no + mul)     
        return sum

File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.js
var totalMoney = function(n) {
    let sum = 0;
    let add = 0;
    let no = 0;
    for(let i = 1; i<=n;i++){
        if (i>n){
            sum += i;
        }
        else{
            no = i % 7;
            if (no == 0){
                sum += 7 + add;
                add++;
            }
            else{
                sum += (add+no)
            }
        }
    }
    return sum;
    
};